pyrtlnet/numpy_inference.py

```diff
@@ -230,32 +231,41 @@ def preprocess_image(self, test_image: np.ndarray) -> np.ndarray:
         #
         # Adding input_zero_point (-128) effectively converts the uint8 image data to
         # int8, by shifting the range [0, 255] to [-128, 127].
-        return np.reshape(
-            test_image / self.input_scale + self.input_zero, newshape=flat_shape
-        ).astype(np.int8)
 
-    def run(self, test_image: np.ndarray) -> tuple[np.ndarray, np.ndarray, int]:
-        """Run quantized inference on a single image.
+        """Taking test_batch of shape (batch_size, 12, 12), each 2D matrix
+        of shape (12,12) is flattened to a 1D column vector of shape (144,),
+        resulting in test_batch's shape becoming (batch_size, 144). Then, we
+        transpose, returning the final shape (144, batch_size), where there
+        are batch_size amount of column vectors of shape (144,), each representing
+        one image.
+        """
+        return test_batch.reshape(test_batch.shape[0], -1).transpose()
+
+    def run(self, test_batch: np.ndarray) -> tuple[np.ndarray, np.ndarray, int]:
+        """Run quantized inference on a batch.
 
         All calculations are done with NumPy and fxpmath.
 
-        :param test_image: An image to run through the NumPy inference implementation.
-
-        :returns: ``(layer0_output, layer1_output, predicted_digit)``, where
-                  ``layer0_output`` is the first layer's raw tensor output, with shape
-                  ``(18, 1)``. ``layer1_output`` is the second layer's raw tensor
-                  output, with shape ``(10, 1)``. Note that these layer outputs are
-                  transposed compared to :func:`.run_tflite_model`. ``predicted_digit``
-                  is the actual predicted digit. ``predicted_digit`` is equivalent to
-                  ``layer1_output.flatten().argmax()``.
+        :param test_batch: A batch of shape (batch_size, 12, 12)
+        to run through the NumPy inference implementation.
```

Comment:
This line needs to be indented 4 more spaces so Sphinx understands that it's a continuation of the `test_batch`'s description, rather than the start of a new paragraph. See the [generated documentation for your pull request](https://pyrtlnet--5.org.readthedocs.build/en/5/numpy_inference.html#pyrtlnet.numpy_inference.NumPyInference.run) and the other `:param:` directives in this file